cli/pom.xml

```diff
@@ -136,11 +136,33 @@
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-surefire-plugin</artifactId>
-                <configuration>
-                    <argLine>-Dspring.config.additional-location=file://${user.home}/.l10n/config/cli/
-                        -Dspring.profiles.active=${user.name},test -Xmx1024m
-                    </argLine>
-                </configuration>
+                <executions>
```

Comment:
I needed to introduce the separate executions here as with the previous config the tests hung when run as part of the maven build. I suspect it has something to do with the AspectJ @Configurable annotation not being setup in the test spring boot container as initially I was getting unsatisfied dependency errors when creating the new object, then when I used MockBeans and extended from CLITestBase the test hung on the RepoUpdateCommandTest so I put the notification tests in their own separate execution.

Reply:
I see. I don't think it is really maintainable to have profile just for a specific test. Not sure what you mean by "then when I used MockBeans and extended from CLITestBase". But iterating on what you were saying, 2 track to investigate:

1) Can change define a test configuration similar to what you did in S3FallbackImageServiceTest?

2) Or (but that would pollute the code), you can make @Autowired(required = false) in ExtractionCheckNotificationSenderSlack and just add them after the instance is created. The problem with that is the runtime code doesn't ensure the dependencies are present, hence some potential NPE that would need to be handled.

3) could refactor not to use @configurable, and just pass everything clearly via the sender constructor

 

I guess what was happening is that @Autowired by default is "required = true" and in that case you basically want to instantiate w

Reply:
Managed to get this back into a single execution profile by drawing inspiration from S3FallbackImageServiceTest so thanks for the suggestion! Issue is a container issue as suspected, there's no DifferentialRevision, SlackClient or SlackChannels beans in the container so AspectJ fails to inject them on creation of a new notifier object. 

I'm still unsure as to why @MockBean annotated on the fields didn't work on its own, but the fix was that I needed to annotate the test class with **@SpringBootTest** and add a static **@TestConfiguration** class that manually created the MockBeans in the container. I then needed to add the test class itself as a class loaded into the container using the **@SpringBootTest** annotation to ensure the TestConfiguration was picked and this allowed tests to pass in the maven build